sympy/core/numbers.py

```diff
@@ -1946,7 +1943,7 @@ class Integer(Rational):
     __slots__ = ['p']
 
     def _as_mpf_val(self, prec):
-        return mlib.from_int(self.p, prec)
+        return mlib.from_int(self.p, prec, rnd)
```

Comment:
this change makes the polygon test fail.  If that test is so fragile that changing the rounding mode causes it to fail, then I'm inclined to blame the test ;-)

A nice kind of fuzzing in numerical analysis is to change the IEEE rounding mode and rerun the code.  Might be nice to try that in SymPy (mpmath really) and see how many of our tests break!

Reply:
It seems that the test was changed. The [original test](https://github.com/sympy/sympy/commit/c5f0b00f#diff-c5cd274236c00ee31386e2953a14a438L1651) looked more robust to me. But maybe something made that fail too.